### PR TITLE
chore: relocate tm2/std.MemFile in gnovm/std

### PR DESCRIPTION
**Pull request by moul**
This PR removes "gno.land" from all `tm2/.../*.go` files.

### CODE REVIEW

**Comment by moul** [comment]
I chose to create `gnovm/pkg/std`, but we should consider merging it with `gnovm/pkg/stdlibs/std`. What do you think?

**Comment by thehowl** [comment]
I have a more general question - why do we name anything `std`? Is there any standard involved?

**Comment by moul** [comment]
> I have a more general question - why do we name anything std? Is there any standard involved?

Right now, we refer to these as "lowest level types." The closest name we have is "sys," and the system contracts are currently named "r/sys." We also considered naming them "r/std." Essentially, this serves as our standard for low-level system components.

In this repository, we have three standard packages designed to be imported to connect with the components without loading other parts (lightweight, no side effects).

An alternative for the Go side would be to rename them to "types." However, it's important to distinguish between "system/low-level types" and "all types." One invites connection with a minimal API (communication layer), while the other is more internal and advanced.

I'm open to reconsidering a renaming, although I believe we will likely prefer to stick with the current approach. My question is about balancing independence with unifying gnovm's standard messages.

**Comment by moul** [comment]
By the way, the relocation is complete, and the CI is green. I have two other PRs in mind that will depend on merging this one. Please feel free to review it, and we can discuss the "naming" topic later, maybe.

**Comment by moul** [comment]
Discussing the naming with @gfanton.

Another suitable location for `MemFile` and `MemPackage` could be directly in the `gnovm/pkg/gnolang` package, as they are frequently used by gnovm. However, since we are considering splitting this package, the question arises: "If the type was initially implemented in gnovm, what would be a good location if we split it?"

File: gno.land/pkg/sdk/vm/msgs.go
Comment (mvertes) [inline gno.land/pkg/sdk/vm/msgs.go:12]: Why this change?

Comment (moul) [reply gno.land/pkg/sdk/vm/msgs.go:12]: probably my editor.

**Comment by zivkovicmilos** [comment]
Let me take a peek at this before merging please 🙏

**Comment by gfanton** [comment]
I also feel that `std` can be confusing due to its use everywhere.

@moul, what do you think about simply prefixing the package name (not the structure, as mentioned in the review meeting) like `std` to `stdsys` (or std-sys), for example? Even if it seems redundant in the package path, it could enhance file and import readability. 

In any case, I don't believe we should block this PR for this, as it's not a regression. Let's open an issue so we can revisit this later.

**Review by thehowl — APPROVED** [review]

**Comment by moul** [comment]
TODO: relocate to `gnovm/types.go`.

**Review by thehowl — APPROVED** [review]

**Comment by jaekwon** [comment]
This should move back to tm2. nothing belongs in the gnovm dir.

**Pull request merged**
